File: agent-driven-architecture-main/examples/kotlin/block/analysis/adapter/src/main/kotlin/adr/blocks/analysis/Adapter.kt
// ── blocks/analysis/adapter — the ONLY file in this block that may hold a store ─
// G13 in one file name (11.2): if you want to know what this block can touch in the
// world, this is the only place to look. It is bound at the single composition
// root, where `wireApp` hands it the relay the other tier reads from.

package adr.blocks.analysis

import adr.spine.pure.Timestamp

/**
 * The rim of the block. `append` stands in for the relay store — a real one would be
 * a Kafka producer, an outbox table or an S3 prefix, held here and nowhere else.
 */
/** The relay's write seam, named rather than a raw `(Timestamp, String) -> Unit`. */
fun interface AppendConclusion {
    operator fun invoke(at: Timestamp, text: String)
}

class LiveRelayWriter(private val append: AppendConclusion) : AnalysisRelay {
    override fun publish(at: Timestamp, text: String) {
        append(at, text)
    }
}

File: agent-driven-architecture-main/examples/typescript/src/blocks/analysis/adapter/adapter.ts
// ── blocks/analysis/adapter/adapter — THIS BLOCK'S IMPURE UNIT (G13) ───────
// The relay handle lives here and nowhere else. In a real deployment this is a
// log-structured store, a topic or an append-only table; here it is a supplied
// writer, so the demo and the tests run offline.

import type { Timestamp } from "@adr/spine/pure/ids";
import type { AnalysisRelay } from "../port";

export function liveRelay(write: (at: Timestamp, text: string) => void): AnalysisRelay {
  return {
    publish: (at: Timestamp, text: string) => write(at, text),
  };
}
